File: 2/rb/rock_paper_scissors.rb
require './outcomes.rb'

def run()
    file_data = read_file("rps.txt")

    puts calculate_score(file_data)
    puts calculate_desired_score(file_data)
end

def read_file(path)
    file_data = File.read(path).split("\n")
end

def calculate_score(data)
    total = 0

    data.each do |round|
        total += outcomes[round]
    end

    total
end

def calculate_desired_score(data)
    total = 0

    data.each do |round|
        total += desired_outcomes[round]
    end

    total
end

File: 2/py/rock_paper_scissors.py
from outcomes import outcomes, desired_outcomes

def run(): 
    stratergy_guide = read_file_spaceless('rps.txt')

    stratergy_points = calculate_score(stratergy_guide)
    print(stratergy_points)
    desired_stratergy_points = calculate_desired_score(stratergy_guide)
    print(desired_stratergy_points)

def read_file(path):
    output = []

    with open(path) as file:
        for line in file:
            output.append(line.strip())

    return output

def read_file_spaceless(path):
    output = []

    with open(path) as file:
        for line in file:
            output.append(line.strip().replace(" ", ""))

    return output

def calculate_score(game):
    total = 0;

    for round in game:
        total += outcomes[round]

    return total

def calculate_desired_score(game):
    total = 0;

    for round in game:
        total += desired_outcomes[round]

    return total
